Source/CarthageKit/GitURL.swift

```diff
@@ -47,21 +47,27 @@ public struct GitURL {
 
 	/// The name of the repository, if it can be inferred from the URL.
 	public var name: String? {
-		let absoluteURLString: String
-		if urlString.hasPrefix(".") {
-			absoluteURLString = URL(fileURLWithPath: FileManager.default.currentDirectoryPath)
-				.appendingPathComponent(urlString)
-				.standardizedFileURL
-				.absoluteString
-		} else {
-			absoluteURLString = urlString
-		}
-		let components = absoluteURLString.split(omittingEmptySubsequences: true) { $0 == "/" }
-
-		return components
+		let lastComponent = urlString
+			.replacingOccurrences(of: "\u{0000}", with: "\u{2400}") // can’t have those
+			.split(omittingEmptySubsequences: true) { $0 == "/" }
 			.last
 			.map(String.init)
-			.map(strippingGitSuffix)
+
+		precondition(lastComponent != .some("")) /* because of `omittingEmptySubsequences` */
```

Comment:
sure we want this in release? `name` is optional

Reply:
I'll address the optionality first — (then, your deeper point about when `urlString` is empty):

If there’s no last path component (e.g. `urlString` is `//` · ⧼still a pathological case⧽), then `lastComponent` is `nil` and the precondition doesn't get hit.

For when `urlString` is empty (or similar whitespace), then in regular Carthage-the-tool usage, our Cartfile-from-NSScanner usage should preclude that from being hit by error-ing first with something like `Parse error: empty or unterminated string after dependency type in line: git "	" "master"`.

So, all good 👍